tmail-backend/tmail-third-party/open-ai/src/main/java/com/linagora/tmail/mailet/AIBotConfig.java

```diff
@@ -37,7 +35,7 @@ public class AIBotConfig {
     private final MailAddress gptAddress;
     private final LlmModel llmModel;
 
-    public AIBotConfig(String apiKey, @Nullable URL baseURL, MailAddress gptAddress, LlmModel llmModel) {
+    public AIBotConfig(String apiKey, MailAddress gptAddress, LlmModel llmModel, @Nullable URL baseURL) {
```

Comment:
Why @Nullable? 

Can we please avoid using null and rely on a strong type (ie a monad) like `Optional` to represent the absence of something?

Reply:
That would mean we need to pass an Optional as an argument? I did that but it gets highlighted in yellow in IntelliJ saying it is against best practices.

Reply:
> That would mean we need to pass an Optional as an argument? 

Better than `null`

>  I did that but it gets highlighted in yellow in IntelliJ saying it is against best practices.

I know Intellij tend to do warning about this. Alternatives might be to create more subtypes.

Yet to me, having something absent is perfectly fine, and handling it through the type system is 1000x better.

Please disregard this IDE warning (unless if we can come up with a cleaner deasign)

Reply:
Don't think we can get a cleaner design, passing an Optional then.